File: DheerajKumar/problem28.cpp
//28. WAP to input a number from user and print its factorial.

#include<iostream>
using namespace std;

int factorial(int number){
    int previous = 1, secondPrevios = 1, current = 1;
    for(int i = 2; i <= number; i++){
        current = previous * i;
        previous = current;
    }
    return current;
}

int main(){
    int number;
    cout<<"Enter the number -> ";
    cin>>number;

    cout<<"The factorial is -> "<<factorial(number);
}

File: LovelyKumari/BasicProgramming/Problem28.java
import java.util.Scanner;

public class Problem28 {
    //WAP to input a number from user and print its factorial.
    public static void main(String[] args) {
        Scanner sc=new Scanner(System.in);
        System.out.print("Enter a number: ");
        int num=sc.nextInt();

        long factorial=1;

        //Check if number is negative
        if(num<0){
            System.out.println("Factorial is not defined for negtaive numbers.");

        }else{
            for(int i=1;i<=num;i++){
                factorial *= i ;
            }
            System.out.println("The factorial of " + num + " is: " + factorial);
        }
        
    }
    
}
